tests/access_tests.py

```diff
@@ -147,44 +175,132 @@ def test_override_role_permissions_drops_absent_perms(self):
             'datasource_access',
             updated_override_me.permissions[0].permission.name)
 
+    def test_cleanup(self):
+        session = db.session
+        TEST_ROLE_1 = 'test_role1'
+        TEST_ROLE_2 = 'test_role2'
+        sm.add_role(TEST_ROLE_1)
+        sm.add_role(TEST_ROLE_2)
+
+        # Case 1. Gamma and gamma2 requested test_role1 on energy_usage access
+        # Gamma already has role test_role1
+        # Extend test_role1 with energy_usage access for gamma2
+        # Check if access request for gamma at energy_usage was deleted
+
+        # gamma2 and gamma request table_role on energy usage
```

Comment:
could u please make all cases - separate functions, they look pretty independent.
I need to refactor mine too.